src/provisioning/Provisioning.Library/Service/TechnicalUserCreation.cs

```diff
@@ -62,6 +62,9 @@ public class TechnicalUserCreation(
 
         var userRoleData = await GetAndValidateUserRoleData(userRolesRepository, userRoleIds).ConfigureAwait(ConfigureAwaitOptions.None);
         var dimConfigRoles = _settings.DimUserRoles.SelectMany(x => x.UserRoleNames.Select(userRoleName => (x.ClientId, userRoleName)));
+        var userProviderRoles = _settings.UserRolesAccessibleByProviderOnly.SelectMany(x => x.UserRoleNames.Select(userRoleName => (x.ClientId, userRoleName)));
+
+        var userTypeId = !(userRoleData.IntersectBy(userProviderRoles, providerData => (providerData.ClientClientId, providerData.UserRoleText)).Any() && technicalUserTypeId == TechnicalUserTypeId.MANAGED) ? technicalUserTypeId : TechnicalUserTypeId.PROVIDER_OWNED;
```

Comment:
Could you please change it to the following due to better readability:

```suggestion
        var userTypeId = technicalUserTypeId == TechnicalUserTypeId.MANAGED && userRoleData.IntersectBy(userProviderRoles, providerData => (providerData.ClientClientId, providerData.UserRoleText)).Any() ? TechnicalUserTypeId.PROVIDER_OWNED : technicalUserTypeId;
```

Reply:
@phil changes down working properly in local build failed here